From b34889553290524a8ad9351d2fd46716986b80f7 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 24 Feb 2021 09:59:46 +0200
Subject: [PATCH] [SW-30985] gaudi2_arc_gnu: asic: change phy FW function

This patch modifies the PHY FW download logic to
suit Gaudi2, based on the new Public Key size and
COLD RST conditions

Change-Id: Ib0f586f7d4fa7ee04a6508f611911bf34a4ae63e
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi2_arc_gnu/asic.c b/gaudi2_arc_gnu/asic.c
index 818e386..f49b74c 100644
--- a/gaudi2_arc_gnu/asic.c
+++ b/gaudi2_arc_gnu/asic.c
@@ -79,14 +79,14 @@
  * ---------------
  * CRC32 [4B]
  * Signature [384B]
- * Public Key [392B]
+ * Public Key [792B]
  * PCIe PHY FW
  * PPboot
  * PPboot size [4B]
  * FW Version [4B]
  * Layout Size [4B]
  */
-#define FW_SRAM_META_SIZE	(4 + 4 + 4 + 392 + 384 + 4)
+#define FW_SRAM_META_SIZE	(4 + 4 + 4 + 792 + 384 + 4)
 void hl_init_pcie_phy_fw(void)
 {
 	int i;
@@ -98,7 +98,8 @@
 	u32 *dst = (u32 *)PCIE_FW_SRAM_BASE;
 	unsigned int rst_src = regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
 
-	if (rst_src != HL_COLD_RST)
+	/* If not cold reset, dont download PHY FW */
+	if (!(rst_src & BIT(HL_COLD_RST)))
 		return;
 
 	for (i = 0; i < (size / 4); i++)
